frontend/src/tests/components/BasicCourseSearch/BasicCourseSearchForm.test.js

```diff
@@ -2,124 +2,180 @@ import React from "react";
-
-		const selectQuarter = getByLabelText("Quarter");
-		userEvent.selectOptions(selectQuarter, "20211");
-		const selectSubject = getByLabelText("Subject Area");
-		userEvent.selectOptions(selectSubject, "ANTH");
-		const selectLevel = getByLabelText("Course Level");
-		userEvent.selectOptions(selectLevel, "G");
-
-		const submitButton = getByText("Submit");
-		userEvent.click(submitButton);
-
-		await waitFor(() => expect(setCourseJSONSpy).toHaveBeenCalledTimes(1));
-		await waitFor(() => expect(fetchJSONSpy).toHaveBeenCalledTimes(1));
-
-		expect(setCourseJSONSpy).toHaveBeenCalledWith(sampleReturnValue);
-		expect(fetchJSONSpy).toHaveBeenCalledWith(
-			expect.any(Object),
-			expectedFields
-		);
-
-	});
-
-	test("when I click submit when JSON is EMPTY, setCourse is not called!", async () => {
-		const sampleReturnValue = {
-			sampleKey: "sampleValue",
-			total: 0,
-		};
-
-		const setCourseJSONSpy = jest.fn();
-		const fetchJSONSpy = jest.fn();
-
-		fetchJSONSpy.mockResolvedValue(sampleReturnValue);
-
-		const { getByText, getByLabelText } = render(
-			<BasicCourseSearchForm
-				setCourseJSON={setCourseJSONSpy}
-				fetchJSON={fetchJSONSpy}
-			/>
-		);
-
-		const selectQuarter = getByLabelText("Quarter");
-		userEvent.selectOptions(selectQuarter, "20204");
-		const selectSubject = getByLabelText("Subject Area");
-		userEvent.selectOptions(selectSubject, "MATH");
-		const selectLevel = getByLabelText("Course Level");
-		userEvent.selectOptions(selectLevel, "G");
-
-		const submitButton = getByText("Submit");
-		userEvent.click(submitButton);
-
-		await waitFor(() => expect(setCourseJSONSpy).toHaveBeenCalledTimes(0));
-		expect(toast).toHaveBeenCalledWith(
-			"If search were implemented, we would have made a call to the back end to get courses for x subject, x quarter, x level",{
-				appearance: "error",
-		});
-	});
-
-
+  const axiosMock = new AxiosMockAdapter(axios);
+  beforeEach(() => {
+    axiosMock
+      .onGet("/api/currentUser")
+      .reply(200, apiCurrentUserFixtures.userOnly);
+    axiosMock
+      .onGet("/api/systemInfo")
+      .reply(200, systemInfoFixtures.showingNeither);
+  });
+  const queryClient = new QueryClient();
+  const addToast = jest.fn();
+  beforeEach(() => {
+    toast.mockReturnValue({
+      addToast: addToast,
+    });
+  });
+
+  test("renders without crashing", () => {
+    render(
+      <QueryClientProvider client={queryClient}>
+        <MemoryRouter>
+          <BasicCourseSearchForm />
+        </MemoryRouter>
+      </QueryClientProvider>
+    );
+  });
+
+  test("when I select a quarter, the state for quarter changes", () => {
+    const { getByLabelText } = render(
+      <QueryClientProvider client={queryClient}>
+        <MemoryRouter>
+          <BasicCourseSearchForm />
+        </MemoryRouter>
+      </QueryClientProvider>
+    );
+    const selectQuarter = getByLabelText("Quarter");
+    userEvent.selectOptions(selectQuarter, "20204");
+    expect(selectQuarter.value).toBe("20204");
```

Comment:
We should probably use QYY format for this instead of YYYYQ.

Reply:
So the YYYYQ format is internal only, when the quarters pop up on the dropdown they are in the QQY format. This is probably because the API works in the YYYYQ format and so everything internally works on YYYYQ.